src/Mailgun/Mailgun.php

```diff
@@ -29,13 +30,16 @@ class Mailgun{
 
     /**
      * @param string|null $apiKey
+     * @param HttpClient $httpClient
      * @param string $apiEndpoint
-     * @param string $apiVersion
-     * @param bool $ssl
      */
-    public function __construct($apiKey = null, $apiEndpoint = "api.mailgun.net", $apiVersion = "v3", $ssl = true){
+    public function __construct(
+        $apiKey = null,
+        HttpClient $httpClient = null,
+        $apiEndpoint = 'api.mailgun.net'
+    ) {
         $this->apiKey = $apiKey;
-        $this->restClient = new RestClient($apiKey, $apiEndpoint, $apiVersion, $ssl);
+        $this->restClient = new RestClient($apiKey, $apiEndpoint, $httpClient);
     }
```

Comment:
I agree, API key should be required. I did not want to make too many BC breaking changes. That's why I left is as it was. 

Thank you for your review.